### ISSUE TITLE
Logstash 2.2 falsely claims worker count was changed by user

### ISSUE THREAD

**Issue by jsvd** [elastic/logstash#4443]
```
/tmp/logstash-2.2.0.snapshot2 % bin/logstash -e "input { stdin {} } filter { multiline { pattern => 'hey' what => previous negate => true } }"
{:pipeline_workers=>4, :pipeline_batch_size=>125, :pipeline_batch_delay=>5, :pipeline_id=>"base"}
Settings: User set pipeline workers: 4
Warning: Manual override - there are filters that might not work with multiple worker threads {:worker_threads=>4, :filters=>["multiline"], :level=>:warn}
```

This is because [agent.rb has a default value for filter workers](https://github.com/elastic/logstash/blob/2.2/logstash-core/lib/logstash/agent.rb#L27) pulled from `LogStash::Pipeline::LOGSTASH_DEFAULTS`.
Then the [pipeline.rb code checks if the setting was passed](https://github.com/elastic/logstash/blob/2.2/logstash-core/lib/logstash/pipeline.rb#L89-L110) (overriden by user), which will always be true.

**Comment by purbon** [comment]
Good catch @jsvd. As I understand the only way to do a manual override is through the ```-w``` param, right? So I guess here the ones who should show this message is the agent, and just check if we passed this parameter? makes sense?

**Comment by jsvd** [comment]
Doing this check before Clamp is very hacky, it would involve grepping ARGV to find either -w N, --filter-workers N, --filter-workers=N.

After Clamp, the two options here are:
a) moving the override check closer to the agent.rb Clamp::Command
b) pass down the distinction between default settings and user defined settings to the pipeline

Both options above assume clamp can queried about the declared options, this needs to be checked.

**Comment by purbon** [comment]
I don't think B makes sense, should be the agent the one who deals with Clamp options and actually report with this kind of changes. When I said checking for ```-w``` i was actually thinking on your option a. What do you think?

**Comment by jsvd** [comment]
closed by #4466
